src/coreComponents/physicsSolvers/fluidFlow/IsothermalCompositionalMultiphaseBaseKernels.hpp

```diff
@@ -2445,6 +2454,54 @@ void KernelLaunchSelector2( integer const numComp, integer const numPhase, ARGS
   }
 }
 
+template< typename KERNELWRAPPER, typename ... ARGS >
+void KernelLaunchSelector_NC_NP_THERM( integer const numComp, integer const numPhase, integer const isThermal, ARGS && ... args )
```

Comment:
why did you have to add this? didn't we already have multiphase compostional thermal kernels?

Reply:
is this ever used? I can't find it...

Reply:
It seems that we have too many of these dispatches. We should only have 1 dispatch for compNumber, phaseNumber and thermal and use it everywhere.

Reply:
will delete... yes we have several dispatches..In the spirit of " too many"  1)  derivative offset structs...  since prop derivatives have thermal slot at index 1.   traits classes  templated on thermal are needed in isothermal runs...  where should we put these structs...     . I think these and dispatches should all be in some common location so the developer knows what is available...   2) further single/multiphase should be combined, and 3) specializations should only be for different variable sets or implicitness  and phase comp partitioning arrays introduced ....

Reply:
why not using the ones you created?

Reply:
In. KernelLaunchSelector2 there is a comment // Ideally this would be inside the dispatch, but it breaks on Summit with GCC 9.1.0 and CUDA 11.0.3.   I thought it would be better to make that change in different PR.